sw/device/tests/dif/dif_otbn_sanitytest.c

```diff
@@ -0,0 +1,252 @@
+  uint32_t start_address = func->func - func->app->imem_start;
+  LOG_INFO("Calling function at address 0x%x on OTBN.", start_address);
+  CHECK(dif_otbn_start(&otbn, start_address) == kDifOtbnOk);
+}
+
+/**
+ * Busy wait for OTBN to be done with its operation.
+ */
+static void otbn_wait_for_done() {
+  bool busy = true;
+
+  while (busy) {
+    CHECK(dif_otbn_is_busy(&otbn, &busy) == kDifOtbnOk,
+          "Unable to get busy status from OTBN");
+  }
+}
+
+/**
+ * Initialize OTBN's data memory with zeros
+ */
+static void zero_dmem() {
+  const uint32_t zero = 0;
+  int dmem_size_words = dif_otbn_get_dmem_size_bytes(&otbn) / 4;
+  for (int i = 0; i < dmem_size_words; ++i) {
+    dif_otbn_result_t rv =
+        dif_otbn_dmem_write(&otbn, i * 4, &zero, sizeof(zero));
+    CHECK(rv == kDifOtbnOk, "Error zeroing word %d in OTBN DMEM: %d", i, rv);
+  }
+}
+
+/**
+ * Run a 384-bit Barrett Multiplication on OTBN and check its result.
+ *
+ * This test is not aiming to exhaustively test the Barrett multiplication
+ * itself, but test the interaction between device software and OTBN. As such,
+ * only trivial parameters are used.
+ *
+ * The code executed on OTBN can be found in sw/otbn/code-snippets/barrett384.s.
+ * The entry point wrap_barrett384() is called according to the calling
+ * convention described in the OTBN assembly code file.
+ */
+static void test_barrett384() {
+  zero_dmem();
+
+  load_app(&kOtbnAppBarrett384);
+
+  // a, first operand
+  const uint8_t a[48] = {10};
+
+  // b, second operand
+  const uint8_t b[48] = {20};
+
+  // m, modulus, max. length 384 bit with 2^384 > m > 2^383
+  // We choose the modulus of P-384: m = 2**384 - 2**128 - 2**96 + 2**32 - 1
+  const uint8_t m[48] = {
+      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xfe, 0xff, 0xff, 0xff, 0xff,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xff, 0xff, 0xff, 0xff};
+
+  // u, pre-computed Barrett constant (without u[384]/MSb of u which is always 1
+  // for the allowed range but has to be set to 0 here).
+  // u has to be pre-calculated as u = floor(2^768/m).
+  const uint8_t u[48] = {
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00,
+      0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x01};
+
+  // c, result, max. length 384 bit.
+  uint8_t c[48] = {0};
+
+  // c = (a * b) % m = (10 * 20) % m = 200
+  const uint8_t c_expected[48] = {200};
+
+  LOG_INFO("Writing input arguments to DMEM");
+  // TODO: Use symbols from the application to load these parameters once they
+  // are available (#3998).
+  CHECK(dif_otbn_dmem_write(&otbn, 0, (uint32_t *)&a, sizeof(a)) == kDifOtbnOk);
```

Comment:
This cast is not allowed, because it violates strict aliasing (assuming that the lvalue `*a` is ever materialized). You will need to rewrite the above constants to be uint32.

Also, could you ad a `/*argument=*/` comment to 0 and other literals in the function call? it's not obvious what those numeric literals mean.